### ISSUE TITLE
How can I use a Vue component as a DecorationWidget?

### ISSUE THREAD

**Issue by kshitizshankar** [scrumpy/tiptap#672]
Hey - Trying to figure out if Decoration Widget can be used with a Vue Component. There was an issue which mentioned it but it never got addressed https://github.com/scrumpy/tiptap/issues/145

Any suggestions on how I could go about doing this? Basically, I am creating a Drag/Drop for FileUpload using the Prosemirror's Upload Example (https://github.com/ProseMirror/website/blob/master/example/upload/index.js) where I can add decorations as markers to show some placeholder/progress stuff in the editor while the file gets upload. I have everything set up but I can't seem to figure out how to use a Vue component in there.

Thanks!
____________

Update 1 - Got a very basic thing working - will be exploring it further to see if I can actually update data in there (using Tiptap's ComponentView class as reference).

```
export const createUploadMarker = (pos, fileId, component) => {
  const decorations = []
  const Component = Vue.extend(component)
  const marker = new Component({
    propsData: { id: id}
  }).$mount()
  const element = 
  decorations.push(
    Decoration.widget(pos, marker.$el, {
      id: id,
      key: id
    })
  )
  return decorations
}
```

**Comment by Alecyrus** [comment]
If you can implement it with vanilla JS, then vanilla JS... Vue component might be overkill ....

**Comment by kshitizshankar** [comment]
@Alecyrus I got it working with the Vue component. For my use-case, there were a lot of things from the existing Vue app that gets utilized within the widget (Sockets, Vuex, Auth, etc) so I just stuck to the Vue component approach as it seemed like the most straightforward. I did explore a bit with the DOM API - but I didn't get anywhere with it (Think I ended up injecting the Vue instance into it which created far more problems so just left it at that)

Overall, it seems to be working fine so far - will close this for now.
